### PR TITLE
feat(generate.js): update for compatibility

### PR DESCRIPTION
**Pull request by TimmyExogenous**
## description

This PR updates the `generate.js` to be compatible with the PR regarding genesis export. Please take a look at the related PR for more details.
[genesis exporting PR](https://github.com/ExocoreNetwork/exocore/pull/108)

This PR hasn't been tested so it might have some issues. 

## Summary by CodeRabbit

- **New Features**
	- Enhanced functionality for managing and processing blockchain-related data.
	- Introduced address generation and key joining capabilities.
	- Improved handling of operator assets and their states.
	- Dynamic association of vaults with whitelisted tokens in the contract.
	- Added support for accurately processing large integers in JSON data.
	- Updated environment variable for clarity regarding its usage.

- **Bug Fixes**
	- Refined logic for calculating withdrawable amounts to improve accuracy.

- **Documentation**
	- Updated data structures and output representation in the genesis file for better clarity.

### CODE REVIEW

File: script/generate.js [hunk 1]
```diff
@@ -254,6 +275,69 @@ async function updateGenesisFile() {
     });
     genesisJSON.app_state.assets.deposits = deposits;
 
+    // x/assets: assets state of the operators
+    const validatorCount = await myContract.methods.getValidatorsCount().call();
+    const operator_assets = [];
+    for (let i = 0; i < validatorCount; i++) {
```
Comment (bwhour) [inline script/generate.js:281]: we can parallelize asynchronous operations to improve processing speed for operator_assets

> const operator_assets = await parallelLimit(xxx)

Comment (TimmyExogenous) [reply script/generate.js:286]: Thanks for the reminder, this script is just used for Bootstrap, and the number of iterations is relatively small. We can use the parallel mechanism when the script can't meet the requirement.

File: .env.example [hunk 2]
```diff
@@ -1,4 +1,4 @@
-SEPOLIA_RPC=http://localhost:8646
+CLIENT_CHAIN_RPC=http://localhost:8646
```
[the remark this answers is not in the document]
Comment (MaxMustermann2) [reply .env.example:1]: Resolved, please check again.

**Review by MaxMustermann2 — APPROVED** [review]
Tested on localnet and generated the `genesis.json` from the integration test. Ran a network of 3 operators, and all good.

Made some minor changes in the last 4 commits.

**Review by trestinlsd — APPROVED** [review]
looks good

**Review by adu-web3 — APPROVED** [review]

**Review by bwhour — APPROVED** [review]
looks good.

File: script/generate.js [hunk 3]
```diff
@@ -276,26 +360,34 @@ async function updateGenesisFile() {
...
-      if (!isValidBech32(opAddressBech32)) {
-        console.log(`Skipping operator with invalid bech32 address: ${opAddressBech32}`);
+      if (!isValidBech32(opAddressExo)) {
+        console.log(`Skipping operator with invalid bech32 address: ${opAddressExo}`);
         continue;
       }
-      const operatorInfo = await myContract.methods.validators(opAddressBech32).call();
-      const operatorCleaned = {
-        earnings_addr: opAddressBech32,
+      const operatorInfo = await myContract.methods.validators(opAddressExo).call();
```
[the remark this answers is not in the document]
Comment (TimmyExogenous) [reply script/generate.js:381]: The program will abort if the call fails.

File: script/generate.js [hunk 4]
```diff
@@ -314,26 +406,37 @@
...
-      for(let j = 0; j < supportedTokens.length; j++) {
+      for (let j = 0; j < supportedTokens.length; j++) {
         const tokenAddress =
           (await myContract.methods.getWhitelistedTokenAtIndex(j).call()).tokenAddress;
+        const selfDelegationAmount = await myContract.methods.delegations(opAddressHex, opAddressExo, tokenAddress).call();
+        amount = amount.plus(
+          new Decimal(selfDelegationAmount.toString()).
+            div('1e' + decimals[j]).
+            mul(exchangeRates[j].toString())
+        );
```
[the remark this answers is not in the document]
Comment (TimmyExogenous) [reply script/generate.js:435]: The divisor won't be zero.

**Pull request merged**